experiment/run_experiment.py

```diff
@@ -242,15 +266,114 @@ def copy_resources_to_bucket(config_dir: str, config: Dict):
     gsutil.rsync(config_dir, destination)
 
 
-class Dispatcher:
-    """Class representing the dispatcher instance."""
+class BaseDispatcher:
+    """Class representing the dispatcher."""
+
+    def __init__(self, config: Dict):
+        self.config = config
+        self.instance_name = experiment_utils.get_dispatcher_instance_name(
+            config['experiment'])
+        self.process = None
+
+    def create_async(self):
+        """Creates the dispatcher asynchronously."""
+        raise NotImplementedError
+
+    def start(self):
+        """Start the experiment on the dispatcher."""
+        raise NotImplementedError
+
+
+class LocalDispatcher:
+    """Class representing the local dispatcher."""
 
     def __init__(self, config: Dict):
         self.config = config
         self.instance_name = experiment_utils.get_dispatcher_instance_name(
             config['experiment'])
         self.process = None
 
+    def create_async(self):
+        """Noop in local experiments."""
+
+    def start(self):
+        """Start the experiment on the dispatcher."""
+        shared_volume_dir = os.path.abspath('shared-volume')
+        if not os.path.exists(shared_volume_dir):
+            os.mkdir(shared_volume_dir)
+        shared_volume_volume_arg = '{0}:{0}'.format(shared_volume_dir)
+        shared_volume_env_arg = 'SHARED_VOLUME={}'.format(shared_volume_dir)
+        sql_database_arg = 'SQL_DATABASE_URL=sqlite:///{}'.format(
```

Comment:
why hardcode sqlite? why use different type of database in cloud and locally? if we think about not just local machines, but other platforms, continuing to use postgres might be more generic, especially if we run the db in a docker container. for the local case we could have a script that sets up https://hub.docker.com/_/postgres. wdyt?

Reply:
>why hardcode sqlite?

This was the easiest way to get things up and running quickly. 
I think a full db is probably overkill for locally run experiments, we may not even need it for the service, I only use it because it makes it easier to deal with persistence, multiple experiments, etc.

>why use different type of database in cloud and locally?

We also use sqlite for testing, it seems to work fine as a drop in replacements because of SQLAlchemy.

Reply:
Sure, whatever is the easiest is fine for this CL, however it might simplify things in the long run if we used the same components in all environments/platforms, including e.g. a small local cluster. Especially when the work is distributed on multiple machines/VMs, a db server sounds more appropriate.